### ISSUE TITLE
Gemm operator doesn't support optional C matrix

### ISSUE THREAD

**Issue by mkolod** [onnx/onnx#1196]
There could conceivably be cases of fully connected layers without biases, in which case it would be silly to create an MxN matrix of zeros, or one broadcastable to MxN, just to satisfy the non-optional nature of C. Although it's uncommon for frameworks to support a case like that, the frameworks do support such cases (`Linear(bias=False)` in PyTorch, `FullyConnected(no_bias=True)` in MxNet, etc.). It seems that even if it's not a very common, it's possibly useful if BatchNorm is used with FullyConnected since then the bias is not necessary.

**Comment by mkolod** [comment]
Nevermind, I can just do a matmul then.
